tools/theme/src/Theme.ts

```diff
@@ -495,6 +494,23 @@ export class Theme extends HTMLElement implements ThemeKindProvider {
         }
     }
 
+    static registerComponentFragment(
+        name: ComponentName,
+        styles: CSSResultGroup
+    ): void {
```

Comment:
To solve the potential issue with bundlers that mangle names, and to avoid accumulating styles in the `fragmentsByComponent` map, you could use a `WeakMap<HTMLElement, CSSResultGroup[]>` instead:

```typescript
    static fragmentsByComponent = new WeakMap<HTMLElement, CSSResultGroup[]>();

    // ...
    static private _getComponentFragments(component: HTMLElement): CSSResultGroup[] {
        let fragments = Theme.fragmentsByComponent.get(component);
        if (!Array.isArray(fragments)) {
          Theme.fragmentsByComponent.set(component, (fragments = []));
        }
        return fragments;
    }

    static registerComponentFragment(
        component: HTMLElement,
        styles: CSSResultGroup
    ): void {
        const fragments = Theme._getComponentFragments(component);
        fragments.push(styles);
```

Reply:
Hmmm, for this to work, we'd need an actual HTMLElement instance to pass to `registerComponentFragment`, so that wouldn't work. Component Fragments need to apply to all instances of a given component, hence using the constructor's name as a key.

However, we could switch this to `this.tagName` instead...